returnn/compile.py

```diff
@@ -228,12 +232,16 @@ def __init__(
         :param returnn_config: RETURNN config object
         :param checkpoint: Path to the checkpoint for export
         :param device: target device for graph creation
+        :param input_names: sequence of model output names
+        :param output_names: sequence of model output names
         :param returnn_python_exe: file path to the executable for running returnn (python binary or .sh)
         :param returnn_root: file path to the RETURNN repository root folder
         """
 
         self.returnn_config = returnn_config
         self.checkpoint = checkpoint
+        self.input_names = input_names
```

Comment:
You miss the logic I stated in https://github.com/rwth-i6/returnn/issues/1517:
```suggestion
        input_names = list(returnn_config.config["extern_data"].keys()) if ("extern_data" in returnn_config.config and input_names is None) else input_names
        output_names = list(returnn_config.config["model_outputs"].keys()) if ("model_outputs" in returnn_config.config and output_names is None) else output_names
        self.input_names = input_names
```

Reply:
why can't we leave them `None`?

Reply:
You can, but you should somewhere have the logic from the code I posted.

Reply:
Now I get it. thanks!